### DISCUSSION TITLE
Bugs

### DISCUSSION THREAD

**Discussion by ETCore7** [Acly/krita-ai-diffusion#558] [Ideas]
Thank you for this great plugin. Unfortunately there are some issues.

- In Live mode, pressing the record button twice will result in a CTD. Recording works fine when images are generated in between, but it's impossible to make a stop-motion animation with only specific images. I'd expect hitting record twice would add the current image to the timeline, but Krita crashes. 

- Update via the menu (Tools - Scripts - Import) does not work. Krita asks for the plugin to be overwritten, but after restart the configuration still gives the old version. Updates were also tried after disabling the plugin and restarts, to no avail. The only thing that works is to manually delete the folder, but this can result in having to download all models again

If this is not the proper section please move/remove. Thank you.

**Comment by Acly** [comment]
> I'd expect hitting record twice would add the current image to the timeline, but Krita crashes.

I fixed the crash when no image is generated, but it will not add the last image (there might not be one). I think it's better to use timeline tools directly to add single keyframes rather than the record button. The plugin doesn't have much control over it unfortunately.

> Update via the menu (Tools - Scripts - Import) does not work

If you have the server/models stored in the same folder, it might not work because they're in use. The Scripts > Import will try to delete the plugin folder before copying the new version anyway.
Since 1.15.0 no new installations should put data into the plugin folder, though old ones may still have that structure and the script won't work for them. See warning [here](https://github.com/Acly/krita-ai-diffusion/wiki/Common-Issues#how-do-i-update-to-a-new-version-of-the-plugin) (I think I also linked this somewhere...)

**Reply by ETCore7** [reply]
Thank you very much.

**Comment by ETCore7** [comment]
My apologies to waste precious time. Tonight I tried to update from 1.16.1 to the latest version, it feels like deja vu.

Installing the update via the script in Krita does not work, the config remains at 1.16.1, despite the question whether the existing AI Diffusion should be overwritten. This also happens after deselecting the (old) plugin and restart. Nothing is in use or running while trying to update.

Manually moving the AI Diffusion folder in AppData/Krita and installing the new version kind of works, but when moving back the map 'server' with the models there is a message on the config screen: 'the folder isn't empty, but nothing is installed.' All checkboxes are cleared, and models and controlnets are ready for download.

I have over 100GB of models and such. Having to download and/or configure everything with each update is kind of tedious. If there is no easy solution that would be fine with me, the tool is worth the time, but it would be great if it were possible to simply give paths where the plugin can detect and configure existing models/loras/controlnets without the need for downloading/configuring everything manually.

Thanks for fixing the record button bug. 

To conclude, a small suggestion. When invoking a controlnet in Live Mode the controlnet Scribble is default, but the controlnet Line Art is far more powerful and intuitive since it takes whatever input is on the canvas and translates this to an image/photo with respect to style and composition and colors, rather than a more loose interpretation of a sketch. I hardly use text prompts anymore since Line Art really tries to mimic the canvas. No "magic button" is needed, and turning down controlnet % and/or pumping up image strength gives a lot of useful variations. It's what I expected AI to be like: just sketch and see an exact rendering. I'd probably make Line Art default, and maybe load it by default with a short explanation.

Thank you.

PS, I still had Krita open. Even with the update and warning that nothing was installed the plugin still claims it is 1.16.1.

**Reply by Acly** [reply]
> Installing the update via the script in Krita does not work

Still not sure why exactly (it worked last time I tried it), but also not something I can easily fix because it's part of Krita. I plan to add some kind of auto-update, but haven't had time yet.

Either way you should be able to reuse/update your server installation and models. Make sure to separate the server installation from the plugin:
* plugin should be extracted into `.../AppData/Roaming/krita/pykrita`
* your user files and default server location is in `.../AppData/Roaming/krita/ai_diffusion`

But you can put the server anywhere else. To upgrade replace only the plugin, and keep the rest.

If it absolutely doesn't detect the existing server install you can also do a minimal fresh one and copy the models (`ComfyUI/models` folder) over, no need to redownload. There is no magic there, it just looks for the files.

**Comment by ETCore7** [comment]
Thank you, I will try.

**Comment by ETCore7** [comment]
I give up. The reason why Krita insisted the plugin was already installed is because there is also a Krita AI Diffusion folder in "D:\Apps\Stable Diffusion GRisk GUI\krita-x64-5.1.0\resource\pykrita". After deleting this the import script works, but moving back the server folder in "AppData\Roaming\..." does not work since (according to the plugin) no server or model or controlnet is installed. The install button is disabled because the folder is not empty, and therefor illegal. I guess I simply delete and download everything again.

**Reply by Acly** [reply]
If the folder isn't recognized something is missing from the install. You can still backup the `ComfyUI/models` folder though and skip all optional downloads. Just copy it back after.

**Comment by ETCore7** [comment]
Thank you. I just downloaded and installed the models via the interface, this only took a few minutes.

Edit * _cry_ *, I noticed I missed the new styles. Config says I have 1.16.1 installed.
